Add observability header to doc landing page

bmorelli25 opened the pull request:
There are plans to refresh and redesign the documentation landing page. In the meantime, this PR adds an Observability heading and moves Observability solutions underneath that heading. With our current build script, I've found two different ways to accomplish this. ~I like option two, but I've included screenshots of option one as well.~ After chatting with Gil, option 1 seems like the best path forward.

Closes https://github.com/elastic/observability-dev/issues/70.

### Option 1 (this PR)

1. **Observability heading added to main doc page** https://www.elastic.co/guide/index.html
    <img width="690" alt="Screen Shot 2019-06-20 at 1 53 15 PM" src="https://user-images.githubusercontent.com/5618806/59883662-ad9bdc00-936a-11e9-8df6-47eb518c6f49.png">
2. **Selecting APM brings you to a new APM subpage. From this page you can view Overview and Server docs.**
    <img width="791" alt="Screen Shot 2019-06-20 at 1 47 31 PM" src="https://user-images.githubusercontent.com/5618806/59883668-b096cc80-936a-11e9-931b-729a0965c76c.png">
3. **Clicking on APM Agents brings you to the Agents subpage**
    <img width="796" alt="Screen Shot 2019-06-20 at 1 47 36 PM" src="https://user-images.githubusercontent.com/5618806/59883663-ad9bdc00-936a-11e9-8f4b-f474f88eb529.png">

* Pro: All URLs remain the same
* Con: 3 clicks to actually view Agent docs

---

### Option 2 ~(this PR)~

1. **Observability heading added to main doc page. (Same as above)** https://www.elastic.co/guide/index.html
    <img width="690" alt="Screen Shot 2019-06-20 at 1 53 15 PM" src="https://user-images.githubusercontent.com/5618806/59883662-ad9bdc00-936a-11e9-8df6-47eb518c6f49.png">
2. **All APM content lives on one subpage.**
    <img width="792" alt="Screen Shot 2019-06-20 at 2 10 15 PM" src="https://user-images.githubusercontent.com/5618806/59883660-ad9bdc00-936a-11e9-9ac4-7deb211c1f5b.png">

* Pro: 2 clicks to view all APM docs
* Cons:
    * A URL redirect is needed for the APM Agent landing page:
    `/guide/en/apm/agent/index.html` moves to join all the conent on:
    `/guide/en/apm/index.html`'
    * The layout of the APM page is not pretty

---

End goal will be for this section to look like this:
<img width="687" alt="Screen Shot 2019-06-20 at 3 03 42 PM" src="https://user-images.githubusercontent.com/5618806/59884252-a7a6fa80-936c-11e9-8599-8231eedf3ef3.png">

bmorelli25 commented:
Thoughts @graphaelli / @alvarolobato ?

graphaelli commented:
I like option 1 better from the perspective of sharing a [link](https://www.elastic.co/guide/en/apm/agent/index.html) to all of the agents we currently support.  It's also the first search result for "elastic apm agents".

That said, I understand the appeal of option 2 from a landing page perspective - thoughts on the above?

bmorelli25 commented:
> I like option 1 better from the perspective of sharing a link to all of the agents we currently support. It's also the first search result for "elastic apm agents".

Both are great points. And we do use the current [link](https://www.elastic.co/guide/en/apm/agent/index.html]) quite a bit in our internal documentation. Being the first result in google doesn't hurt either.

This landing page is _hopefully_ going to be going through another redesign sometime in the next 3-12 months. Probably would be better to leave URLs as is and if necessary we can reassess when the permanent solution is ready.

bmorelli25 commented:
Switched to "option 1" and rebased with master to keep `7.2` current. Unable to test this as the doc build is broken right now. Will try again in a bit.

bmorelli25 commented:
New screenshots:

<img width="678" alt="Screen Shot 2019-06-26 at 11 04 19 AM" src="https://user-images.githubusercontent.com/5618806/60204706-52536900-9804-11e9-84dc-cb6360a6570b.png">
<img width="788" alt="Screen Shot 2019-06-26 at 11 04 51 AM" src="https://user-images.githubusercontent.com/5618806/60204703-51bad280-9804-11e9-901c-3cc4fc33df87.png">
<img width="802" alt="Screen Shot 2019-06-26 at 11 18 33 AM" src="https://user-images.githubusercontent.com/5618806/60204702-51bad280-9804-11e9-8dc0-518fe9f478a1.png">

Tagging @jrodewig / @lcawl  - if you have time for a review, it'd be appreciated!
